src/event/reader.rs

```diff
@@ -400,21 +441,38 @@ impl EventReader {
     /// to different Segments of the stream are received. In-case we receive data for an already
     /// acquired SegmentSlice this method waits until SegmentSlice is completely consumed before
     /// returning the data.
-    pub async fn acquire_segment(&mut self) -> Option<SegmentSlice> {
-        info!("acquiring segment for reader {}", self.id);
+    ///
+    /// Note: it may return an error indicating that the reader is not online. This means
+    /// that another thread removes this reader from the ReaderGroup probably because the host of this reader
+    /// is assumed dead.
+    pub async fn acquire_segment(&mut self) -> Result<Option<SegmentSlice>, Error> {
```

Comment:
Humm, should we make our own return value type? Also is io.Error really what we want here? It makes sense for byteStream, but here I am less sure.

Reply:
I have improved the code and introduced a custom error called `EventReaderError`